### PR TITLE
Make EXPORTED_RUNTIME_METHODS work for library functions

### PR DESCRIPTION
**Pull request by sbc100**
This helps when moving function from runtime to JS library since it
means that users who were previously using `EXPORTED_RUNTIME_METHODS` to
export such symbols will not be effected by the transition.

Previously it was also necessary to add such function to
`DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` (with a different mangling!).

This is step towards unifying the way exports work.  See #8380

### CODE REVIEW

File: src/modules.js [hunk 1]
```diff
@@ -344,6 +344,29 @@ function isExportedByForceFilesystem(name) {
          name === 'removeRunDependency';
 }
 
+// When running with ASSERTIONS enabled we create stubs for each library
+// function that that was not included in the build.  This gives usefull errors
```
Comment (kripken) [inline src/modules.js:348]:
```suggestion
// function that that was not included in the build.  This gives useful errors
```

File: src/modules.js [hunk 2]
```diff
@@ -344,6 +344,29 @@ function isExportedByForceFilesystem(name) {
...
+// when library dependencies are missing from `__deps` or depended on without
+// being added to DEFAULT_LIBRARY_FUNCS_TO_INCLUDE
+// TODO(sbc): These errors could potentially be generated at build time via
+// some kind of acron pass that searched for uses of these missing symbols.
```
Comment (kripken) [inline src/modules.js:352]:
```suggestion
// some kind of acorn pass that searched for uses of these missing symbols.
```

File: src/modules.js [hunk 3]
```diff
@@ -344,6 +344,29 @@ function isExportedByForceFilesystem(name) {
...
+function addMissingLibraryStubs() {
+  if (!ASSERTIONS) return '';
+  let rtn = '';
+  const librarySymbolSet = new Set(libraryFunctions);
+  for (const ident in LibraryManager.library) {
+    if (typeof LibraryManager.library[ident] === 'function') {
+      if (ident[0] === '$' && !isJsLibraryConfigIdentifier(ident) && !LibraryManager.library[ident + '__internal']) {
```
Comment (kripken) [inline src/modules.js:359]:
```suggestion
      if (ident[0] === '$' && !isJsLibraryConfigIdentifier(ident)) {
```
`__internal` is handled in that helper function

**Review by kripken — COMMENTED** [review]
Please mention this in settings.js and the changelog.

File: src/modules.js (hunk 3 above)
Comment (sbc100) [reply src/modules.js:359]: This is checking if there exists another entry called `ident + '__internal'`..  isJsLibraryConfigIdentifier only checks `ident` itself.

Comment (sbc100) [reply src/modules.js:359]: I created a helper to make this more clear.

File: tests/test_browser.py [hunk 4]
```diff
@@ -5174,12 +5174,16 @@ def test_wasm_worker_sleep(self):
   # Tests emscripten_terminate_wasm_worker()
   @also_with_minimal_runtime
   def test_wasm_worker_terminate(self):
-    self.btest(test_file('wasm_worker/terminate_wasm_worker.c'), expected='0', args=['-sWASM_WORKERS'])
+    self.set_setting('WASM_WORKERS')
+    self.set_setting('DEFAULT_LIBRARY_FUNCS_TO_INCLUDE', ['$dynCall'])
```
[the remark this answers is not in the document]
Comment (sbc100) [reply tests/test_browser.py:5178]: Split out into #17377

[the remark this answers is not in the document]
Comment (sbc100) [reply tests/test_browser.py]: Rebased

**Review by kripken — APPROVED** [review]

**Pull request merged**